android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -821,6 +824,77 @@ class GeckoEngineSession(
         })
     }
 
+    /**
+     * Convenience method for the error when session translation is not available.
+     */
+    private fun sessionTranslationNotAvailable(): Throwable {
+        val errorMessage = "A translation session coordinator is not available."
+        logger.error(errorMessage)
+        return IllegalStateException(errorMessage)
+    }
+
+    /**
+     * See [EngineSession.requestTranslate]
+     */
+    override fun requestTranslate(
+        fromLanguage: String,
+        toLanguage: String,
+        options: TranslationOptions?,
+    ) {
+        var geckoOptions: GeckoViewTranslateSession.TranslationOptions? = null
+
+        if (options != null) {
+            geckoOptions =
+                GeckoViewTranslateSession.TranslationOptions.Builder()
+                    .downloadModel(options.downloadModel).build()
```

Comment:
I think you can move the initialisation of geckoOptions inside  if (geckoSession.sessionTranslation != null) , because geckoOptions object is useless if geckoSession.sessionTranslation is null.

Reply:
Thanks, updated, great optimization